src/server.c

```diff
@@ -2677,17 +2681,20 @@ void populateCommandStructure(struct redisCommand *c) {
     populateCommandMovableKeys(c);
 
     /* Assign the ID used for ACL. */
-    c->id = ACLGetCommandID(c->name);
+    c->id = ACLGetCommandID(c->fullname);
 
     /* Handle subcommands */
     if (c->subcommands) {
-        for (int j = 0; c->subcommands[j].name; j++) {
+        for (int j = 0; c->subcommands[j].fullname; j++) {
             struct redisCommand *sub = c->subcommands+j;
 
             /* Translate the command string flags description into an actual
              * set of flags. */
-            setImplictACLCategories(sub);
+            setImplicitACLCategories(sub);
             populateCommandStructure(sub);
+            sds old_name = sub->fullname;
+            sub->fullname = catSubCommandFullname(c->fullname, sub->fullname);
+            sdsfree(old_name);
```

Comment:
in order to save the redundant `sdsnew` and `sdsfree` maybe we should initialize the fullname outside of `populateCommandStructure`..? not sure about this, leaving it for you to decide